profiler/profile.go

```diff
@@ -44,48 +44,125 @@ const (
 	MetricsProfile
 )
 
-func (t ProfileType) String() string {
-	switch t {
-	case HeapProfile:
-		return "heap"
-	case CPUProfile:
-		return "cpu"
-	case MutexProfile:
-		return "mutex"
-	case BlockProfile:
-		return "block"
-	case GoroutineProfile:
-		return "goroutine"
-	case expGoroutineWaitProfile:
-		return "goroutinewait"
-	case MetricsProfile:
-		return "metrics"
-	default:
-		return "unknown"
+// collector holds the implementation details of a ProfileType, see collectors
+// map below.
+type collector struct {
+	// Name is used to lookup the profile for pprof.Lookup(name) and returned by
+	// ProfileType.String(). For profile types that don't use this approach, the
+	// name can be chosen arbitrarily.
+	Name string
+	// Filename is the filename used for uploading the profile to the datadog
+	// backend which is aware of them. Delta profiles are prefixed with "delta-"
+	// automatically.
+	Filename string
+	// Delta enables delta profiling for pprof profiles. This is useful for
```

Comment:
Same as the golang/go repository, `bool`s are many times defined using the word "reports". Perhaps it could be better to do that here too?
```suggestion
	// Delta reports whether delta profiling is enabled for this profile. This is useful for
```

Reply:
Arguably the new sentence structure is more complicated and verbose than before, but I don't feel strongly about it, so I'll apply.